src/components/List_Products.js

```diff
@@ -1,53 +1,80 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import { Link, Outlet } from 'react-router-dom';
 import { onSnapshot } from 'firebase/firestore';
 
-import { productsCollection, saveItem } from '../lib/api';
+import { getListFromDB } from '../lib/api';
+import './ListProducts.css';
 
 function ListProducts() {
   const [items, setItems] = useState([]);
   const [itemName, setItemName] = useState('');
+  const list = useRef({});
 
-  /* Get items */
   useEffect(() => {
-    const unsubscribe = onSnapshot(productsCollection, (snapshot) => {
-      let products = [];
+    /* Get token */
+    const token = localStorage.getItem('token');
 
-      snapshot.forEach((doc) => {
-        products.push({ ...doc.data(), id: doc.id });
-      });
-
-      setItems(products);
+    /* Get items */
+    const unsubscribe = onSnapshot(getListFromDB(token), (doc) => {
+      list.current = doc.data();
+      setItems(doc.data().items);
     });
     return () => {
       unsubscribe();
     };
   }, []);
 
-  /* Save Item */
-  const saveItemName = (e) => {
-    e.preventDefault();
-    saveItem(itemName);
-    setItemName('');
-  };
-
   const handleChange = (e) => {
```

Comment:
could be a one liner

Reply:
Sorry I don't understand very well. Could you elaborate?

Reply:
nvm, my bad, I didn't see you were setting state